Shared/Samples/Augment reality to navigate route/AugmentRealityToNavigateRouteView.ARSceneView.swift

```diff
@@ -0,0 +1,292 @@
+                        scene: model.scene,
+                        graphicsOverlays: [model.routeGraphicsOverlay]
+                    )
+                }
+                .calibrationButtonAlignment(.bottomLeading)
+                .onCalibratingChanged { isPresented in
+                    model.scene.baseSurface.opacity = isPresented ? 0.6 : 0
+                }
+                .task {
+                    Task {
+                        do {
+                            try await model.startTrackingLocation()
+                        } catch {
+                            self.error = error
+                        }
+                    }
+                }
+                .overlay(alignment: .top) {
+                    Text(model.statusText)
+                        .multilineTextAlignment(.center)
+                        .frame(maxWidth: .infinity, alignment: .center)
+                        .padding(8)
+                        .background(.regularMaterial, ignoresSafeAreaEdges: .horizontal)
+                }
+                .onAppear {
+                    model.statusText = "Adjust calibration before starting."
+                }
+                .onDisappear {
+                    model.stopNavigation()
+                }
+                Divider()
+            }
+            .toolbar {
+                ToolbarItemGroup(placement: .bottomBar) {
+                    Button("Start") {
+                        isNavigating = true
+                        Task {
+                            do {
+                                try await model.startNavigation()
+                            } catch {
+                                print("Failed to start navigation.")
+                            }
+                        }
+                    }
+                    .disabled(isNavigating)
+                }
+            }
+            .errorAlert(presentingError: $error)
+        }
+    }
+}
+
+extension AugmentRealityToNavigateRouteView {
+    // MARK: Scene Model
+    
+    /// The view model for scene view in the sample.
+    @MainActor
+    class SceneModel: ObservableObject {
+        /// A scene with an imagery basemap.
+        @Published var scene = Scene(basemapStyle: .arcGISImagery)
+        
+        /// The graphics overlay containing a graphic for the route.
+        @Published var routeGraphicsOverlay = GraphicsOverlay()
+        
+        /// The elevation surface set to the base surface of the scene.
+        @Published var elevationSurface: Surface = {
+            let elevationSurface = Surface()
+            elevationSurface.navigationConstraint = .unconstrained
+            elevationSurface.opacity = 0
+            elevationSurface.backgroundGrid.isVisible = false
+            return elevationSurface
+        }()
+        
+        /// The elevation source with elevation service URL.
+        @Published private var elevationSource = ArcGISTiledElevationSource(url: .worldElevationService)
```

Comment:
I could be wrong, but it seems like most of the variables in this class don't need to be `@Published` since they don't drive any state changes, especially  `elevationSource` since it is private.